addons/purchase_product_matrix/static/src/js/purchase_product_field.xml

```diff
@@ -3,16 +3,18 @@
 <templates xml:space="preserve">
 
     <t t-name="purchase.PurchaseProductField" t-inherit="account.ProductLabelSectionAndNoteField" t-inherit-mode="primary">
-        <xpath expr="//button[@t-if='hasExternalButton']" position="before">
-            <button
-                t-if="isConfigurableTemplate"
-                type="button"
-                class="btn btn-link fa fa-pencil px-2"
-                tabindex="-1"
-                draggable="false"
-                t-att-aria-label="configurationButtonHelp"
-                t-att-data-tooltip="configurationButtonHelp"
-                t-on-click="onEditConfiguration"/>
+        <xpath expr="//Many2One" position="after">
+            <t t-if="!props.readonly and isConfigurableTemplate">
```

Comment:
Is it considered a bad practice to have the t-if on the xml element directly in client-side templates (like it was before) ? Or is it your personal preference to keep it separate ?

Reply:
It's a personal preference. It makes it clearer that the button is conditional.
I like to see `t-if` and `t-foreach` like `if` and `for` in js or python.
```
if (condition) {
  button...
}
if condition:
  button
```
I don't considerate it especially as a bad practice, sometimes it's better for xpath sometimes it's worst.